src/marqo/tensor_search/health.py

```diff
@@ -1,10 +1,108 @@
+        numeric_watermark = validation.validate_nonnegative_number(watermark[:-2], "OpenSearch disk watermark value")
+        multiplier = 1024 ** constants.BYTE_SUFFIXES.index(watermark[-2:].lower())
+        return numeric_watermark * multiplier
+        
+    elif watermark[-1].lower() == "b":
+        # Watermark in BYTE format
+        # Bytes represent MIN disk space AVAILABLE
+        numeric_watermark = validation.validate_nonnegative_number(watermark[:-1], "OpenSearch disk watermark value")
+        return numeric_watermark
+    
+    # Percentage or Ratio calculation
+    if watermark[-1] == "%":
+        # Watermark in PERCENTAGE format
+        # Ratio & percentage represent MAX disk space USED
+        numeric_watermark = validation.validate_nonnegative_number(watermark[:-1], "OpenSearch disk watermark value")
+        ratio_watermark = (100 - numeric_watermark) / 100
+    else:
+        # Watermark in RATIO format
+        # Ratio & percentage represent MAX disk space USED
+        numeric_watermark = validation.validate_nonnegative_number(watermark, "OpenSearch disk watermark value")
+        ratio_watermark = 1 - numeric_watermark
+
+    if ratio_watermark < 0 or ratio_watermark > 1:
+        raise errors.InternalError("OpenSearch watermark ratio or percentage cannot be negative or more than 100%.")
+    if total_in_bytes is None:
+        raise errors.InternalError("total_in_bytes must be provided for ratio or percentage watermark.")
+    
+    # Round up to the next byte (more conservative approach)
+    return math.ceil(total_in_bytes * ratio_watermark)
+
+
+def check_opensearch_disk_watermark_breach(config: Config):
+    """
+    Checks if the OpenSearch disk watermark is breached:
+    1. Check disk watermark from the settings endpoint.
+      - Check transient, persistent, then default settings
+      - convert it to a size in BYTES (it could initially be a percentage, ratio, or size in B, GB, MB, etc.)
+    2. Check the current available disk space from the stats endpoint.
+    3. Compare current avilable space to watermark value.
+
+    Returns: red if watermark is breached, green otherwise.
+    """
+
+    # Query opensearch for watermark
+    raw_flood_stage_watermark = None
+    opensearch_settings = HttpRequests(config).get(path="_cluster/settings?include_defaults=true&filter_path=**.disk*")
+    for settings_type in constants.OPENSEARCH_CLUSTER_SETTINGS_TYPES:
+        try:
+            # Check for transient, persistent, then defaults settings
+            opensearch_disk_settings = opensearch_settings[settings_type]["cluster"]["routing"]["allocation"]["disk"]
+            raw_flood_stage_watermark = opensearch_disk_settings["watermark"]["flood_stage"]
+            logger.debug(f"Found disk flood stage watermark in {settings_type} settings: {raw_flood_stage_watermark}")
+            break
+        except KeyError:
+            logger.debug(f"Flood stage watermark not found in {settings_type} settings.")
```

Comment:
Is this expected to happen under normal circumstances? Otherwise you should raise it

Reply:
Yes, this is actually just a debug message, it will indicate every time watermark is not found (let's say in transient or persistent settings), then it will just check the next level of settings. I can remove this debug statement.